lib/css/components/sidebar-widgets.less

```diff
@@ -0,0 +1,245 @@
+.s-sidebarwidget {
+    --_sw-content-px: calc(var(--su16) - var(--su1)); // subtract 1px for border
+    --_sw-content-spacing-inner: var(--su12); // the spacing between two adjacent simple items
+    --_sw-content-spacing-outer: var(--su16); // the spacing at the start/end of a group of simple items, as well as between a complex item and its separator line
```

Comment:
These three custom properties seem like constants. It took me a moment to realize that, but I'd want that difference to be clearer. Naming conventions for custom properties that act as scoped constants is an issue I'd like to solve.

@b-kelly you said you've seen the PPCP approach in the wild. Have you noticed a convention for differentiating constant from variable properties in CSS?

Reply:
What will be the interim alternative here?  Inlining the constants?
I haven't seen PPCP in the wild but if we want to borrow a concept from the JS community I guess we could simply make the word uppercase (e.g. `_SW-CONTENT-PX`)?

Reply:
For the time being I differentiated constants from variables with a comment similar to what has been done already [here](https://github.com/StackExchange/Stacks/blob/develop/lib/css/components/expandable.less#L40). I am happy to leave this conversation open but I believe this should not block the PR from being merged since it goes beyond its initial scope. If we make a decision about it we can document it and add a task to our backlog to rename the constants all at once (it should not be a lot of work if they are already highlighted by a comment).

Reply:
Regarding constants (as in, a reused, static value that should not be changed by others), I'd recommend simply using Less variables to bake it in on the server-side. If we want to remain client-side only, then there's no other way to really make our "pseudo-private" into "true private". One thing I've seen before that may of use to us is to use a double underscore (`--__c-constant-var`) to really underscore (heh) the point that you _really_ shouldn't be changing those.

Of the two options, I prefer the former. This may be a nuance we'd like to record in the ADR.

Reply:
Thanks @b-kelly for the input. I actually remember vaguely having [this conversation](https://github.com/StackExchange/Stacks/pull/1091#issuecomment-1273040356) about using less variables for "constants" a while back already in an older PR. 
I also do prefer the former approach although I remember it was going against the principle of sticking to native CSS as much as possible.
[This article](https://www.smashingmagazine.com/2018/05/css-custom-properties-strategy-guide/#local-static-variables-are-ok-sometimes) from Smashing Magazine support keeping static variable in preprocessor variables too.
Let's wait for @dancormier to be back, make a decision together, update our ADR and potentially create a tech task to go through the codebase and adjust all constants to whatever approach we have decided.

As I said already in my previous comment. This decision should not block the merging of this PR though.

Reply:
Using Less variables is a simple way to differentiate constants from variables and it would be easy for us to change in the future if the need arises. I'm 👍 on that approach.

@giamir do you think this calls for a ticket along the lines of "_use Less variables for component constants_"?

---

**A note on moving away from Less variables (and general dependency on Less)**: I have a ticket for an ADR wrt less reliance on Less but, until that's done, I'll just mention I'm not dogmatic about this. There are and will be exceptions where Less just makes the most sense and this is one of them.

Reply:
> @giamir do you think this calls for a ticket along the lines of "use Less variables for component constants"?

@dancormier Yes, I think that's a good idea. Would you mind adding the ticket? I also believe we could put it under the "Operational Maintenance" epic and prioritize it as any other tech debt/investment. We should also adjust the PPCP ADR to reflect this decision - we can either do it straight away or as part of the ticket. 🙂